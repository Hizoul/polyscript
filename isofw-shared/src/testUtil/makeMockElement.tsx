import * as React from "react"

const makeMockElement = (name: string, showChildren: boolean = false) => {
  return (props: any) => (
    <div className={name}>
      {JSON.stringify({...props, children: undefined}, undefined, 2)}
      {props.children}
    </div>
  )
  }

export default makeMockElement
