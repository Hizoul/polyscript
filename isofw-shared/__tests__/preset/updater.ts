import sharedPresetUpdater from "isofw-shared/src/components/preset/updater";
import presetUpdaterTest from "isofw-shared/src/tests/preset/updater"
import makeMockElement from "isofw-shared/src/testUtil/makeMockElement";

presetUpdaterTest(sharedPresetUpdater(makeMockElement("presetupdater")))
