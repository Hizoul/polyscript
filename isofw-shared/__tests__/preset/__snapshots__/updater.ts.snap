// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preset Update works as planned:  after setting 2 falls  1`] = `
<div
  className="presetupdater"
>
  {
  "id": "c1bbbbbbbbbbbbbbbb098765",
  "item": {
    "_id": "c1bbbbbbbbbbbbbbbb098765",
    "number": 0,
    "project": "-",
    "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    "isReady": false
  },
  "isReady": false
}
</div>
`;

exports[`Preset Update works as planned:  after setting to true  1`] = `
<div
  className="presetupdater"
>
  {
  "id": "c1bbbbbbbbbbbbbbbb098765",
  "item": {
    "_id": "c1bbbbbbbbbbbbbbbb098765",
    "number": 0,
    "project": "-",
    "camera": "c1bbbbbbbbbbbbbbbbbbbbbb",
    "isReady": true
  },
  "isReady": true
}
</div>
`;

exports[`Preset Update works as planned: Before anything 1`] = `
<div
  className="presetupdater"
>
  {
  "id": "c1bbbbbbbbbbbbbbbb098765",
  "isReady": false
}
</div>
`;
